### PR TITLE
INT 1077 add weblinks

### PR DESCRIPTION
**Pull request by VDubber**
Add weblinks property to the alerts and sensors entities.
Formatted project

### CODE REVIEW

File: src/converters.ts [hunk 1]
```diff
@@ -148,7 +152,20 @@ function deviceSensorKey(deviceId: number): string {
   return `cbdefense-sensor-${deviceId}`;
 }
 
+/**
```
Comment (VDubber) [inline src/converters.ts:155]: Core change

File: src/converters.ts [hunk 2]
```diff
@@ -181,40 +201,60 @@ function alertFindingDisplayName(data: CarbonBlackAlert): string {
...
-  ].filter(e => !!e);
+  ].filter((e) => !!e);
   if (components.length !== 3) {
     return data.id;
   }
-  return components.join(" : ");
+  return components.join(' : ');
+}
+
+/**
```
Comment (VDubber) [inline src/converters.ts:211]: Core change

File: src/invocationValidator.ts [hunk 3]
```diff
@@ -27,20 +27,20 @@ export default async function invocationValidator({
   const instanceConfig = instance.config;
 
   if (!instanceConfig) {
-    throw new IntegrationValidationError("Configuration missing");
+    throw new IntegrationValidationError('Configuration missing');
   }
 
   const { site, orgKey, connectorId, apiKey } = instanceConfig;
 
-  if (!(site && orgKey && connectorId && apiKey)) {
```
Comment (VDubber) [inline src/invocationValidator.ts:35]: Core change: removed site

File: src/converters.ts [hunk 4]
```diff
@@ -148,7 +152,20 @@ function deviceSensorKey(deviceId: number): string {
...
 
+/**
+ * Constructs a device sensor web link.
+ * Ex. https://defense-prod05.conferdeploy.net/cb/investigate/events/events?searchWindow=ONE_MONTH&query=device_id%3A6923993
+ * @param site
+ * @param deviceId
+ */
+function deviceSensorWebLink(site: string, deviceId: number): string {
+  const url = new URL(`${siteWeblink(site)}/inventory/endpoints`);
+  url.searchParams.append('s[query]', `${deviceId}`);
```
Comment (adam-in-ict) [inline src/converters.ts:163]: Does the searchParams.append call modify the name to remove the `s` and square brackets?  More for my knowledge than any concern that this is incorrect.  I haven't used the URL module a ton and couldn't find documentation on that particular use (plus the use of searchParams.append and its corresponding example in comments below would include the square brackets in the resulting URL.

File: src/invocationValidator.ts [hunk 5]
```diff
@@ -27,20 +27,20 @@ export default async function invocationValidator({
...
+  if (!(orgKey && connectorId && apiKey)) {
     throw new IntegrationValidationError(
-      "Configuration requires site, orgKey, connectorId, and apiKey",
+      'Configuration requires site, orgKey, connectorId, and apiKey',
```
Comment (adam-in-ict) [inline src/invocationValidator.ts:37]: If we're removing site from the check a few lines up, do we want to also remove it from this error message?

**Review by adam-in-ict — DISMISSED** [review]

File: src/converters.ts (hunk 4 above)
Comment (VDubber) [reply src/converters.ts:163]: The URL does get encoded so the `[` and `]` are replaced with "safe" characters, but the `s` is URL safe already so it stays.
I left the URL example un-encoded for readability.

Here's a real example: `https://defense-prod8.conferdeploy.net/inventory/endpoints?s%5Bquery%5D=7891906`
This URL does work. One of the reasons I waited for an account was so I could test this exact question.

**Review by adam-in-ict — APPROVED** [review]

**Pull request merged**